backend/src/main/java/com/chinguv45tier3team53/tripplannerrest/controller/TransportationController.java

```diff
@@ -30,8 +35,10 @@ public ResponseEntity<Transportation> getTransportation(@PathVariable long id) {
         }
     }
 
-    @PostMapping("/list")
-    public ResponseEntity<Transportation> addTransportation(@RequestBody Transportation transportation) {
+    @PostMapping("/trip/{id}")
+    public ResponseEntity<Transportation> addTransportationToTrip(@PathVariable long id, @RequestBody Transportation transportation) {
+        Trip existingTrip = tripService.getTripById(id);
+        existingTrip.addTransportation(transportation);
         Transportation addedTransportation = service.save(transportation);
```

Comment:
you would also need to persist the changes made to the trip object here, otherwise the transportation wouldn't belong to any trip

Reply:
yeah, so about that...

if I also save the trip using the tripService, the one to many relationship probably won't be established

I had same situations in my previous projects. I initially wrote the code as you suggested, but when I attempted to retrieve the many in one to many relationship, it failed. I also ran sql queries in mysql to see what's going on, but I discovered the one to many relationship wasn't established.

But when I wrote the code this way, it worked. The relationship was established, and I was able to retrieve the many.

Also, I used postman to test the current code. I added transportations to a trip and when I attempted to retrieve all the transportations for the trip, it was successful.

I don't know why this way works, I only have guesses. I did see examples of writing the code this way and here is one of them
https://hellokoding.com/jpa-and-hibernate-one-to-many-unidirectional-spring-boot/

Maybe things have changed now due to the updates in spring framework. I can change the code as you requested and test it using postman if you want me to.